src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -1117,7 +1167,10 @@ private static int FormatOctal(long value, Span<byte> destination)
             ulong remaining = (ulong)value;
             Span<byte> digits = stackalloc byte[32]; // longer than any possible octal formatting of a ulong
 
+            digits.Slice(0, digits.Length - 1).Fill(0x30); // Prefill with '0' chars except the last, which should remain 0x0 (null terminator)
+
             int i = digits.Length - 1;
+
```

Comment:
Here's where the bug was: The buffer should've been prefilled with 0x30 chars.

The while loop will be accessed _once_ when `value` is 0, subtracting the `i` iterator _once_ as well.

Reply:
Nevermind. This was unnecessary.